ef_tests/tests/tests.rs

```diff
@@ -30,35 +38,25 @@ fn execute_test(test: TestUnit) {
     .is_success());
 }
 
-#[cfg(test)]
-mod ef_tests {
-    use std::collections::HashMap;
-
-    use ef_tests::types::TestUnit;
-
-    use crate::execute_test;
-
-    #[test]
-    fn add11_test() {
-        let s: String =
-            std::fs::read_to_string("./vectors/add11.json").expect("Unable to read file");
-        let tests: HashMap<String, TestUnit> =
-            serde_json::from_str(&s).expect("Unable to parse JSON");
+fn parse_test_file(file: &str) -> HashMap<String, TestUnit> {
+    let s: String = std::fs::read_to_string(file).expect("Unable to read file");
+    let tests: HashMap<String, TestUnit> = serde_json::from_str(&s).expect("Unable to parse JSON");
+    tests
+}
 
-        for (_k, test) in tests {
-            execute_test(test)
-        }
+fn parse_and_execute_test_file(file: &str) {
+    let tests = parse_test_file(file);
+    for (_k, test) in tests {
+        execute_test(&test)
     }
+}
 
-    #[test]
-    fn solidity_example_test() {
-        let s: String =
-            std::fs::read_to_string("./vectors/solidityExample.json").expect("Unable to read file");
-        let tests: HashMap<String, TestUnit> =
-            serde_json::from_str(&s).expect("Unable to parse JSON");
+#[cfg(test)]
+mod ef_tests {
+    use crate::parse_and_execute_test_file;
 
-        for (_k, test) in tests {
-            execute_test(test)
-        }
+    #[test]
+    fn beacon_root_contract_calls_test() {
+        parse_and_execute_test_file("./vectors/cancun/eip4788_beacon_root/beacon_root_contract/beacon_root_contract_calls.json");
     }
 }
```

Comment:
We could run them all with [datatest_stable](https://docs.rs/datatest-stable/latest/datatest_stable/), but it doesn't support ignoring tests.

Reply:
nice, I will investigate for the follow up.

Reply:
In MLIR we switched to it after experimenting with [libtest-mimic](https://crates.io/crates/libtest-mimic/0.7.3) ([code here](https://github.com/lambdaclass/evm_mlir/blob/fee2671a39a3c7a1089cdd471e0a79cfeb94b537/tests/ef_tests.rs)).